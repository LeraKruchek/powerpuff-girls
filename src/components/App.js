import React, { Fragment } from 'react';
import "../stylesheets/main.scss";
// App component
export class App extends React.Component {

  // render
  render() {
    // show the loading state while we wait for the app to load
    const {children} = this.props;

    // render
    return (
      <Fragment>
        <div className="content">
          {children}
        </div>
        <div className="footer">
          <img src="//static.tvmaze.com/images/tvm-header-logo.png" />
          <span>
            The Powerpuff Girls website
          </span>
        </div>
      </Fragment>
    );
  }
}

export default App;
