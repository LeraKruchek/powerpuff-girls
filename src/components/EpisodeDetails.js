import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import qs from 'query-string';
import { fetchEpisode } from '../actions/episodes';

class EpisodeDetails extends React.Component {
  
  constructor(props) {
    super(props);

    const query = qs.parse(this.props.location.search);

    this.state = {
      season: query.season,
      number: query.number
    }
  }
  
  componentDidMount() {
    const { episode } = this.props;
    
    if (!episode) {
      this.props.dispatch(fetchEpisode({season: this.state.season, number: this.state.number}));
    }
  }
  
  render() {
    const { episode } = this.props;
    
    if (episode) {
      episode.formattedDate =
        new Date(episode.airdate).toLocaleDateString('en-US',
          { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' });
    }
    
    return (
      <Fragment>
        {
          episode ?
          <Fragment>
            <h1 className="content-header">{episode.name}</h1>
            <div className="container flex flex-column">
              <div className="container flex">
                <img src={episode.image.medium}/>
                <div className="block" dangerouslySetInnerHTML={{
                  __html: episode.summary,
                }}/>
              </div>
              <div className="episode-info">
                <h2>Episode info</h2>
                <p><span className="heading">Number: </span><span>Season {episode.season}, </span><span>Episode {episode.number}</span></p>
                <p><span className="heading">Airdate: </span>{episode.formattedDate} at {episode.airtime}</p>
              <p><span className="heading">Runtime: </span>{episode.runtime} minutes</p>
              </div>
              <Link to="/">Back</Link>
            </div>
          </Fragment> :
          'Loading...'
        }
      </Fragment>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  const query = qs.parse(ownProps.location.search);
  const id = String(query.season) + String(query.number);

  return {
    episode: state.episodes ? state.episodes[id] : null
  }
};

export default connect(mapStateToProps)(EpisodeDetails);
