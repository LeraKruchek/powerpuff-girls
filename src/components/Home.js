import React from "react";
import { connect } from 'react-redux';
import { fetchShow } from '../actions/shows';
import EpisodeDetails from './EpisodeDetails';

// Home page component
class Home extends React.Component {
  constructor(props) {
    super(props);
  }
  
  componentDidMount() {
    this.props.dispatch(fetchShow());
  }
  
  // render
  render() {
    const { show } = this.props;
    return (
      <div className="page-home">
        {
          show ?
          <div>
            <h1>{show.name}</h1>
            <div className="flex flex-column">
              <div className="flex flex-space">
                <img src={show.image.medium}/>
                <div className="block" dangerouslySetInnerHTML={{
                  __html: show.summary,
                }}/>
              </div>
              
              <div className="episodes">
                <EpisodeDetails episodes={show._embedded.episodes}/>
              </div>
            </div>
          </div> :
          'Loading...'
        }
      </div>
    );
  }
}

// export the connected class
function mapStateToProps(state) {
  return {
    show: state.show,
  };
}
export default connect(mapStateToProps)(Home);
