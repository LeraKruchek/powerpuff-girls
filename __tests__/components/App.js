import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { mountToJson } from 'enzyme-to-json';
import { App } from '../../src/components/App';

configure({ adapter: new Adapter() });
// unit tests for the App component
describe('App component', () => {
  describe('render()', () => {
    it('should render the component', () => {
      const props = {};
      const wrapper = shallow(<App {...props} />);
      expect(wrapper.length).toEqual(1);
      expect(mountToJson(wrapper)).toMatchSnapshot();
    });
  });
});
