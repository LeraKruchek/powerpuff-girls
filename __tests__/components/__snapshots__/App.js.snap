// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component render() should render the component 1`] = `
<Fragment>
  <div
    className="content"
  />
  <div
    className="footer"
  >
    <img
      src="//static.tvmaze.com/images/tvm-header-logo.png"
    />
    <span>
      The Powerpuff Girls website
    </span>
  </div>
</Fragment>
`;
